clang/test/Driver/sycl-post-link-options.cpp

```diff
@@ -0,0 +1,22 @@
+// REQUIRES: system-linux
+/// Verify same set of sycl-post-link options generated for old and new offloading model
+// Test for JIT compilation
+// RUN: %clangxx --target=x86_64-unknown-linux-gnu -fsycl --offload-new-driver \
+// RUN:          -Xdevice-post-link -O0 -v %s 2>&1 \
+// RUN:   | FileCheck -check-prefix OPTIONS_POSTLINK_JIT %s
+// RUN: %clangxx --target=x86_64-unknown-linux-gnu -fsycl --no-offload-new-driver \
+// RUN:          -Xdevice-post-link -O0 -v %s 2>&1 \
+// RUN:   | FileCheck -check-prefix OPTIONS_POSTLINK_JIT %s
+// OPTIONS_POSTLINK_JIT: sycl-post-link{{.*}} -O0 -O2 -device-globals -spec-const=native -split=auto -emit-only-kernels-as-entry-points -emit-param-info -symbols -emit-exported-symbols -split-esimd -lower-esimd
+
+#include <sycl/sycl.hpp>
+using namespace sycl;
+
+int main(void) {
+  sycl::queue queue;
+  sycl::event event = queue.submit([&](sycl::handler &cgh) {
+    cgh.parallel_for<class set_range>(sycl::range<1>{16},
+                                      [=](sycl::id<1> idx) {});
+  });
+  return 0;
+}
```

Comment:
These two tests don't look particularly different.  At a high level, these don't need to be scrutinized with the `REQUIRES:` restriction and should work as a single test with just different `--target` values to exercise the given target.

Reply:
The test failed under windows. So, I split it into two.

Reply:
I have two tests. But I have removed the need to compile them (replaced with dry runs)